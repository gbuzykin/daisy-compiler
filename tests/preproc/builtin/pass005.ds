#define SUM(a, b) ((a) + (b))
#define SUM2(a, b, c) SUM  (SUM (a, c), SUM   (b, c))
#define LINE_PLUS_SUM2(a, b, c) ((__line__) + SUM2(a, b, c))

const v = __str__(LINE_PLUS_SUM2  (1, 2, 3) ## SUM2  (1, 2, 3));
const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
