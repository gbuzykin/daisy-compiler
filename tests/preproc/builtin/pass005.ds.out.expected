./preproc/builtin/pass005.ds:5:1: debug: token
 5 | const v = __str__(LINE_PLUS_SUM2  (1, 2, 3) ## SUM2  (1, 2, 3));
   | ^~~~~
./preproc/builtin/pass005.ds:5:7: debug: id: v
 5 | const v = __str__(LINE_PLUS_SUM2  (1, 2, 3) ## SUM2  (1, 2, 3));
   |       ^
./preproc/builtin/pass005.ds:5:9: debug: token
 5 | const v = __str__(LINE_PLUS_SUM2  (1, 2, 3) ## SUM2  (1, 2, 3));
   |         ^
./preproc/builtin/pass005.ds:5:11: debug: string: "((5) + ((((1) + (3))) + (((2) + (3)))))((((1) + (3))) + (((2) + (3))))"
 5 | const v = __str__(LINE_PLUS_SUM2  (1, 2, 3) ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:5:64: debug: token
 5 | const v = __str__(LINE_PLUS_SUM2  (1, 2, 3) ## SUM2  (1, 2, 3));
   |                                                                ^
./preproc/builtin/pass005.ds:6:1: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   | ^~~~~
./preproc/builtin/pass005.ds:6:7: debug: id: v
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |       ^
./preproc/builtin/pass005.ds:6:9: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |         ^
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 6
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 1
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 3
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 2
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 3
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 1
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 3
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 2
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: integer number: 3
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:11: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./preproc/builtin/pass005.ds:6:71: debug: token
 6 | const v = __paste__(LINE_PLUS_SUM2  (1, 2, 3) ## + ## SUM2  (1, 2, 3));
   |                                                                       ^
./preproc/builtin/pass005.ds: info: warnings 0, errors 0
